pkg/reconciler/managed/reconciler.go

```diff
@@ -675,6 +686,19 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 			managed.SetConditions(v1alpha1.ReconcileError(errors.Wrap(err, errReconcileCreate)))
 			return reconcile.Result{RequeueAfter: r.shortWait}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 		}
+		log.Debug("Successfully requested creation of external resource", "requeue-after", time.Now().Add(r.shortWait))
+		record.Event(managed, event.Normal(reasonCreated, "Successfully requested creation of external resource"))
+
+		if creation.ResourceLateInitialized {
+			// We use APIUpdatingApplicator to enforce the update to go through
+			// by setting object's resourceVersion to the latest.
+			if err := resource.NewAPIUpdatingApplicator(r.client).Apply(ctx, managed); err != nil {
```

Comment:
I'd prefer us to have this entire reconciler use a `resource.ClientApplicator` instead of a `client.Client` that we can mock out rather than explicitly creating a `NewAPIUpdatingApplicator` inline.